scripts/helpers/build.js

```diff
@@ -50,27 +75,37 @@ module.exports = (options = {}) => {
     bundler.bundle();
   });
 
-  const babelPromise = new Promise((resolve, reject) => {
-    spawn("babel", [libInDir, "--out-dir", libOutDir], {
-      stdio: "inherit"
-    }).on("exit", code => {
-      if (code) {
-        reject();
-      } else {
-        resolve();
-      }
-    });
+  // ignore alloy.js because it will be built separately below.
+  const babelPromise = run("babel", [
+    libInDir,
+    "--out-dir",
+    libOutDir,
+    "--ignore",
+    alloyInFile
+  ]);
 
-    if (watch) {
-      spawn(
-        "babel",
-        [libInDir, "--out-dir", libOutDir, "--watch", "--skip-initial-build"],
-        {
-          stdio: "inherit"
-        }
-      );
-    }
-  });
+  if (watch) {
+    run("babel", [
+      libInDir,
+      "--out-dir",
+      libOutDir,
+      "--watch",
+      "--skip-initial-build",
+      "--ignore",
+      alloyInFile
+    ]);
+  }
+
+  // The package resolution in launch does not follow dependencies on npm packages, so we need to build our
+  // own Alloy file here.
+  // I tried having rollup run Babel, but it didn't respect the .browserslistrc located in the src/lib directory.
+  const alloyPromise = fsPromises
+    .mkdir(tempDir)
+    .catch(() => undefined)
+    .then(() => fsPromises.copyFile(browserslistrcFile, browserslistrcTempFile))
+    .then(() => run("rollup", ["-c"]))
+    .then(() => run("babel", [alloyTempFile, "--out-dir", libOutDir]))
+    .finally(() => toPromise(callback => rimraf(tempDir, callback)));
```

Comment:
My understanding is that Babel doesn't attempt to transpile anything in node_modules by default. That might explain why having rollup run babel without being inside the temp directory wasn't working.

Babel talks about transpiling node_modules here: https://babeljs.io/docs/en/configuration#whats-your-use-case
We might need to change our .babelrc to babel.config.json. We probably should anyway.

Whether we try to do this now or later is up to you.

Reply:
Are you saying we wouldn't need to use rollup if babel could just transform the files in node_modules?

Reply:
No, I think I'm just addressing your comment:

> I tried having rollup run Babel, but it didn't respect the .browserslistrc located in the src/lib directory.

I got the feeling that you had to deal with creating the `temp` directory and running `rollup` and `babel` as separate operations because you couldn't get rollup to run babel while properly transpiling Alloy from node_modules. I think if we change `.babelrc` to `babel.config.json`, then rollup running babel might properly transpile the code, which I think would allow us to also not need this `temp` directory. In other words, my thought is that it's not that babel wasn't respecting `.browserlistrc`, but that babel wasn't running on Alloy from node_modules at all.

I certainly could have misunderstood though.